server/src/services/trainingLinks.service.ts

```diff
@@ -24,124 +27,172 @@ interface ILinks {
+    lieu_formation_geo_coordonnees: 1,
+    rome_codes: 1,
+    _id: 0,
+  }
+) => FormationCatalogue.findOne(query, filter)
 
-  // tentative de récupération des données à partir d'une catalogueFormation
-  if (training.cle_ministere_educatif) {
-    formations = await FormationCatalogue.findOne({ cle_ministere_educatif: training.cle_ministere_educatif }, { lieu_formation_geo_coordonnees: 1, rome_codes: 1, _id: 0 })
-    if (formations) {
-      return buildLbaLinkFromFormation(formations)
-    }
+/**
+ * @description get formation according to the available parameters passed to the API endpoint
+ * @param {object} wish wish data
+ * @returns {Promise<IFormationCatalogue>}
+ */
+const getTrainingFromParameters = (wish: IWish) => {
+  // search by cle ME
+  if (wish.cle_ministere_educatif) {
+    return getFormation({ cle_ministere_educatif: wish.cle_ministere_educatif })
   }
 
-  // pas de formation trouvée. on utilise les autres paramètres fournis pour construire le lien lbac
-  const lbaLink = new URL(`${lbaDomain}/recherche-emploi`)
+  // search by uai_lieu_formation
+  if (wish.uai_lieu_formation) {
+    return getFormation({ $or: [{ cfd: wish.cfd }, { rncp_code: wish.rncp }], uai_lieu_formation: wish.uai_lieu_formation })
+  }
+
+  // search by uai_formateur
+  if (wish.uai_formateur) {
+    return getFormation({ $or: [{ cfd: wish.cfd }, { rncp_code: wish.rncp }], uai_formateur: wish.uai_formateur })
+  }
+
+  // search by uai_formateur_responsable
+  if (wish.uai_formateur_responsable) {
+    return getFormation({ $or: [{ cfd: wish.cfd }, { rncp_code: wish.rncp }], uai_formateur_responsable: wish.uai_formateur_responsable })
+  }
+}
 
-  // identification des romes
-  let romes = null
-  formations = await FormationCatalogue.find(
+/**
+ * @description get training booking link for a specific training
+ * @param {object} wish wish data
+ * @returns {Promise<string>} LBA link
+ */
+const getPrdvLink = async (wish: IWish): Promise<string> => {
+  if (!wish.cle_ministere_educatif) {
+    return ""
+  }
+
+  const elligibleFormation = await EligibleTrainingsForAppointment.findOne(
     {
-      $or: [
-        {
-          rncp_code: training.rncp,
-        },
-        {
-          cfd: training.cfd,
-        },
-        {
-          "bcn_mefs_10.mef10": training.mef,
-        },
-      ],
+      cle_ministere_educatif: wish.cle_ministere_educatif,
+      lieu_formation_email: {
+        $ne: null,
+        $exists: true,
+        $not: /^$/,
+      },
     },
-    {
-      rome_codes: 1,
-      _id: 0,
-    }
-  ).limit(5)
-
-  if (formations.length) {
-    const romeSet = new Set()
-    formations.map(({ rome_codes }) => {
-      rome_codes.map((rome) => romeSet.add(rome))
-    })
-    romes = Array.from(romeSet)
-  }
+    { _id: 1 }
+  )
+
+  return buildEmploiUrl({
+    baseUrl: `${config.publicUrlEspacePro}/form`,
+    params: { referrer: "lba", cleMinistereEducatif: encodeURIComponent(wish.cle_ministere_educatif), ...utmData },
```

Comment:
```suggestion
    params: { referrer: "lba", cleMinistereEducatif: wish.cle_ministere_educatif, ...utmData },
```
`url.searchParams.append(key, value)` dans `buildEmploiUrl` encode déjà les caractères spéciaux